src/cookie-io.js

```diff
@@ -0,0 +1,226 @@
+import { defaultData } from "./default-data.js";
+import { CONSTANTS } from "./constants/constants.js";
+
+function getDefaultData() {
+    // Reset data (deep copy)
+    return JSON.parse(JSON.stringify(defaultData));
+}
+
+class CharacterData {
+    characterData = getDefaultData();
+
+    getName() {
+        return this.characterData.name;
+    }
+    setName(name) {
+        this.characterData.name = name;
+    }
+
+    addGold(amount) {
+        if (this.characterData.gold + amount >= 0) {
+            this.characterData.gold += amount;
+        } else {
+            console.log(
+                "Error: tried to set gold to negative number, amount:",
+                amount,
+                "current gold:",
+                this.characterData.gold
+            );
+        }
+    }
+    getGold() {
+        return this.characterData.gold;
+    }
+
+    getCurrentLevel() {
+        return this.characterData.currentLevel;
+    }
+    setCurrentLevel(level) {
+        if (this.checkScene(level)) {
+            this.characterData.currentLevel = level;
+        }
+    }
+
+    getTotalEnemiesKilled() {
+        let sum = 0;
+        for (const level in this.characterData.levels) {
+            for (const enemy in this.characterData.levels[level].enemiesKilled) {
+                sum += this.characterData.levels[level].enemiesKilled[enemy];
+            }
+        }
+        return sum;
+    }
+
+    addTimesClicked(value) {
+        this.characterData.timesClicked += value;
+    }
+    getTimesClicked() {
```

Comment:
https://medium.com/@luke_smaki/javascript-es6-classes-8a34b0a6720a

you can do like 

get timesClicked() and then access it like character.timesClicked to make it more like other types of objects in javascript.

Reply:
Didn't know they had static functions and stuff like that js, nice